core/connection/manager.go

```diff
@@ -252,17 +244,14 @@ func (manager *connectionManager) consumeOpenvpnStates(stateChannel <-chan openv
 	log.Debug(managerLogPrefix, "State updater stopped")
 }
 
-func (manager *connectionManager) onStateChanged(state openvpn.State, sessionID session.SessionID) {
+func (manager *connectionManager) onStateChanged(state State, sessionID session.SessionID) {
 	manager.mutex.Lock()
 	defer manager.mutex.Unlock()
 
 	switch state {
-	case openvpn.ConnectedState:
-		manager.statsKeeper.MarkSessionStart()
```

Comment:
I am a bit concerned with a move of manager.statsKeeper.MarkSessionStart() to openvpn package, since it should be as a general contract to implement. Specific transport should always implement statistics collection. Most basic assumption is that each connection will have a start and a finish. Any implementation details should be hidden behind those.